### PR TITLE
MULE-15657: Add <additionalPluginDependencies> to Mule Maven Plugin

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: mule-packager/src/main/java/org/mule/tools/api/classloader/model/resolver/AdditionalPluginDependenciesResolver.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    Map<BundleDependency, List<BundleDependency>> pluginsWithAdditionalDeps = new LinkedHashMap<>();
+    for (Plugin pluginWithAdditionalDependencies : pluginsWithAdditionalDependencies) {
+      BundleDependency pluginBundleDependency = getPluginBundleDependency(pluginWithAdditionalDependencies, mulePlugins);
+      ClassLoaderModel pluginClassLoaderModel =
+          getPluginClassLoaderModel(pluginWithAdditionalDependencies, mulePluginsClassLoaderModels);
+      List<BundleDependency> additionalDependencies = new ArrayList<>();
+      pluginWithAdditionalDependencies.getAdditionalDependencies().stream()
+          .filter(additionalDep -> !isPresentInClassLoaderModel(pluginClassLoaderModel, additionalDep))
+          .forEach(
+                   dep -> {
```
Comment (WilliamPinho) [inline mule-packager/src/main/java/org/mule/tools/api/classloader/model/resolver/AdditionalPluginDependenciesResolver.java]: Please extract this to a method

File: mule-packager/src/main/java/org/mule/tools/api/classloader/model/resolver/AdditionalPluginDependenciesResolver.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      if (!additionalDependencies.isEmpty()) {
+        pluginsWithAdditionalDeps.put(pluginBundleDependency,
+                                      additionalDependencies);
+      }
+    }
+    return pluginsWithAdditionalDeps;
+  }
+
+  private BundleDependency getPluginBundleDependency(Plugin plugin, List<BundleDependency> mulePlugins) {
+    return mulePlugins.stream().filter(mulePlugin -> mulePlugin.getDescriptor().getArtifactId().equals(plugin.getArtifactId())
```
Comment (WilliamPinho) [inline mule-packager/src/main/java/org/mule/tools/api/classloader/model/resolver/AdditionalPluginDependenciesResolver.java]: Please use StringUtils.equals

File: mule-packager/src/main/java/org/mule/tools/api/classloader/model/resolver/AdditionalPluginDependenciesResolver.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                                                            .getGroupId().equals(plugin.getGroupId())
+                                                            && pluginClassLoaderModel.getArtifactCoordinates().getArtifactId()
+                                                                .equals(plugin.getArtifactId()))
+        .findFirst()
+        .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find ClassLoaderModel resolved for plugin: "
+            + plugin)));
+  }
+
+  private boolean areSameArtifact(Dependency dependency, Artifact artifact) {
+    return dependency.getArtifactId().equals(artifact.getArtifactCoordinates().getArtifactId())
```
Comment (WilliamPinho) [inline mule-packager/src/main/java/org/mule/tools/api/classloader/model/resolver/AdditionalPluginDependenciesResolver.java]: Please use StringUtils.equals

File: mule-packager/src/main/java/org/mule/tools/api/classloader/model/util/PluginUtils.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Utils for {@link Plugin} construction.
+ *
+ * @since 3.2.0
+ */
+public class PluginUtils {
+
+  public static List<Plugin> toPluginDependencies(Map<BundleDependency, List<BundleDependency>> pluginsAndDependencies) {
+    return pluginsAndDependencies.entrySet().stream().map(
+                                                          (pluginEntry) -> {
```
Comment (WilliamPinho) [inline mule-packager/src/main/java/org/mule/tools/api/classloader/model/util/PluginUtils.java]: Please extract this to a method

File: mule-artifact-it/mule-packager-it/src/test/resources/repository/group/id/b/artifact-id-b/1.0.0-SNAPSHOT/_remote.repositories [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#NOTE: This is a Maven Resolver internal implementation file, its format can be changed without prior notice.
+#Thu Sep 20 10:31:28 ART 2018
+artifact-id-b-1.0.0-SNAPSHOT.jar>=
+artifact-id-b-1.0.0-SNAPSHOT.pom>=
```
Comment (gsfernandes) [inline mule-artifact-it/mule-packager-it/src/test/resources/repository/group/id/b/artifact-id-b/1.0.0-SNAPSHOT/_remote.repositories]: Remove this file

File: mule-artifact-it/mule-packager-it/src/test/resources/repository/group/id/b/artifact-id-b/1.0.0-SNAPSHOT/artifact-id-b-1.0.0-SNAPSHOT.jar.lastUpdated [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#NOTE: This is a Maven Resolver internal implementation file, its format can be changed without prior notice.
```
Comment (gsfernandes) [inline mule-artifact-it/mule-packager-it/src/test/resources/repository/group/id/b/artifact-id-b/1.0.0-SNAPSHOT/artifact-id-b-1.0.0-SNAPSHOT.jar.lastUpdated]: Same here, remove this file

File: mule-classloader-model/src/main/java/org/mule/tools/api/classloader/ClassLoaderModelJsonSerializer.java [hunk 7]
```diff
@@ -53,15 +54,17 @@ public static ClassLoaderModel deserialize(File classLoaderModelDescriptor) {
...
+        .create();
     ClassLoaderModel parameterizedClassloaderModel = classLoaderModel.getParametrizedUriModel();
     return gson.toJson(parameterizedClassloaderModel);
   }
 
   /**
    * Serializes the classloader model to the classloader-model.json file in the destination folder
    *
-   * @param classLoaderModel the classloader model of the application being packaged
+   * @param classLoaderModel  the classloader model of the application being packaged
```
Comment (gsfernandes) [inline mule-classloader-model/src/main/java/org/mule/tools/api/classloader/ClassLoaderModelJsonSerializer.java]: remove this whitespace added

**Pull request merged**